libraries/AP_Motors/AP_MotorsMatrix.cpp

```diff
@@ -246,9 +246,14 @@ void AP_MotorsMatrix::output_armed_stabilizing()
     // ensure that throttle_avg_max is between the input throttle and the maximum throttle
     throttle_avg_max = constrain_float(throttle_avg_max, throttle_thrust, throttle_thrust_max);
 
-    // throttle providing maximum roll, pitch and yaw range
-    // calculate the highest allowed average thrust that will provide maximum control range
-    float throttle_thrust_best_rpy = MIN(0.5f, throttle_avg_max);
+    // Ensure desired altitude output is not sacrificed for attitude output, but allow sacrifice of climb rate.
+    float desired_altitude_output = MIN(throttle_thrust, get_throttle_hover() * compensation_gain);
+
+    // throttle facilitating physically maximum roll, pitch or yaw output.
+    constexpr float THROTTLE_FOR_MAX_RPY{0.5f};
```

Comment:
even if it seems stupid to document numerical figures, i think its best to do so because it immediately takes all mystery out of them _for everyone_. number of lines is unimportant (assuming no effects during compilation), and clarity should be optimized for. but i leave it to consensus of ardupilot devs. i can change this to whatever ardupilot devs prefer, just point me to the style guide, etc.